mongo/integration/unified/collection_operation_execution.go

```diff
@@ -1009,9 +1009,14 @@ func executeListIndexes(ctx context.Context, operation *operation) (*operationRe
 	return newCursorResult(docs), nil
 }
 
-func executeRenameCollection(ctx context.Context, operation *operation) (*operationResult, error) {
+func executeRename(ctx context.Context, operation *operation) (*operationResult, error) {
 	coll, err := entities(ctx).collection(operation.Object)
+	// Rename operations can also be run against GridFS buckets. Check for a bucket entity of
+	// the same name and run executeBucketRename if an entity is found.
 	if err != nil {
+		if _, bucketOk := entities(ctx).gridFSBucket(operation.Object); bucketOk == nil {
+			return executeBucketRename(ctx, operation)
+		}
```

Comment:
The conflation of collection and GridFS bucket rename is kinda confusing. This logic seems to make more sense to put in the `operation.run()` switch statement.

E.g.
```go
case "rename":
	// "rename" can either target a collection or a GridFS bucket.
	if _, err := entities(ctx).collection(op.Object); err == nil {
		return executeRenameCollection(ctx, op)
	}
	if _, err := entities(ctx).gridFSBucket(op.Object); err == nil {
		return executeBucketRename(ctx, op)
	}
	return fmt.Errorf("failed to find a collection or GridFS bucket named %q", operation.Object
// ...
```

Reply:
Good idea. Moved and used very similar code to that above; thanks!